zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -573,6 +600,11 @@ cdef arrays_from_rows(DatetimeIndex_t dates,
             out_shape,
             size,
             ts_ixs,
+            (
+                all_rows[AD_FIELD_NAME].values.view('int64')
+                if len(all_rows) else
```

Comment:
Might be worth noting that we're working around pandas losing dtype metadata for empty objects here.

Reply:
done